### ISSUE TITLE
Crash with @article in Spanish

### ISSUE THREAD

**Issue by bruno-maltras** [alex-ball/biblatex-oxref#18]
Thank you very much for the style (I love it for items with orig-date). 

Issue: When the bibliography contains a reference of @article type (in the .bib file), it crashes if the language is set to Spanish (is ok in English) with this error:

Use of \\blx@imc@printtext doesn't match its definition.
Use of \es@dots doesn't match its definition.
TeX capacity exceeded, sorry [input stack size=5000].

I have been able to check only the previous version, as I cannot access to an install with the  
3.0.

**Comment by alex-ball** [comment]
Could you provide a minimal working example (short .tex file, .bib file with an entry that triggers the bug)?

I suspect there may be an incompatibility with a package I don't currently have installed.

**Comment by bruno-maltras** [comment]
I am using Lyx, it is not pure latex.

**Comment by bruno-maltras** [comment]
Here is the export to .tex by Lyx

(edited - Sorry, uploaded via github)

**Comment by alex-ball** [comment]
I don't think you can post files to GitHub via email. Please could you log into the site and upload them?

**Comment by bruno-maltras** [comment]
Here you have:

[Ox-ref_Error_Article_Spanish.zip](https://github.com/alex-ball/biblatex-oxref/files/10978025/Ox-ref_Error_Article_Spanish.zip)

**Comment by alex-ball** [comment]
Okay, I've located the issue.

**TLDR answer:**

Until I change the approach, here is a workaround you can use in your preamble *after* loading an oxref style:

```tex
\makeatletter
\AtBeginDocument{%
  \def\blx@ox@abbrevstring{%
    ABCDEFGHIJKLMNOPQRSTUVWXYZ.}%
}
\makeatother
```

**Detailed answer:**

In the manual it says "If the last/only word of the journal title is abbreviated (indicated by point or capital letter), it does not need a comma after it. Oxref will try to detect this[...]". To do this, it checks to see if the last letter of the journaltitle (or journalsubtitle, if provided) is a capital letter or a dot. I couldn't find a built-in test for this, so the style checks to see if the last letter is in the string:

```tex
\newcommand*{\blx@ox@abbrevstring}{%
  A\^{A}BCDEFG\u{G}HI\.{I}\^{I}JKLMNO\"{O}\^{O}PQRSTU\"{U}\^{U}VWXYZ.}%
```

The accents `\c{C}` and `\c{S}` are added `\AtBeginDocument` if the font encoding isn't OT1.

The thing that causes the crash is the `\.{I}` accent. Even though `babel-spanish` defines `\...` so as to let you continue to use `\.` for an overdot diacritic, it does appear to break `xstring`. Here is a MWE that demonstrates this:

```tex
\documentclass{standalone}
\usepackage[spanish]{babel}
\usepackage{xstring}
\begin{document}
Test of \.{I}: \IfSubStr{\.{I}}{\.{I}}{Yes}{No}.
\end{document}
```

Removing the `\.{I}` accent from the test string avoids triggering this bug, which is how the workaround above works.

I think the proper solution for `oxref` is to stick to ASCII capitals and introduce a new command along the lines of `\DeclareNumChars` and `\DeclareRangeChars`.

**Comment by alex-ball** [comment]
By the way, using `oxref` in Spanish is likely to give you missing bibliography string errors, so if you would be willing to provide translations, please open another issue and I'll set up a branch where you can add them in.

**Comment by bruno-maltras** [comment]
Thanks a lot Alex. I really appreciate your work.
I will be using the workaraund in the meanwhile. And I will be more than happy to help with Spanish translations for strings. I open a new issue

Bruno
